pkg/manager/devicehandler.go

```diff
@@ -40,12 +41,12 @@ func (m *Manager) DeviceDisconnected(id topodevice.ID, err error) error {
 	}, backoff.NewExponentialBackOff())
 }
 
-func updateDevice(deviceStore device.Store, id topodevice.ID, connectivity topodevice.ConnectivityState, channel topodevice.ChannelState,
-	service topodevice.ServiceState) error {
-	topoDevice, err := deviceStore.Get(id)
+func updateDevice(deviceStore device.Store, id topodevice.ID, connectivity topodevice.ConnectivityState, channel topodevice.ChannelState, service topodevice.ServiceState) error {
+	object, err := deviceStore.Get(topo.ID(id))
 	if err != nil {
 		return err
 	}
+	topoDevice := topo.ObjectToDevice(object)
```

Comment:
I think it is a bit of a concern that `ObjectToDevice` does not give any error - there could be may things wrong with the object - e.g. no address, unknown kind, no version - because we're running off an attribute map, there are many possibilities

Reply:
@SeanCondon - I plan to move ObjectToDevice and DeviceToObject functions from onos-topo to onos-config (since Device is only used by onos-config now). Will add validation as part of that change. Currently, onos-topo allows creation of "anonymous" entities (i.e. entities with no Kind). Need to add that check in onos-topo as well.

Reply:
https://github.com/onosproject/onos-config/issues/1163

Reply:
It would be worth investigating if `Device` can be removed from `onos-config` altogether - it could be replaced globally with `Entity`